backend/src/test/java/com/ez/ezbackend/budget/request/ScheduleRequestTest.java

```diff
@@ -0,0 +1,156 @@
+  }
+
+  @Test(expected = EzIllegalRequestException.class)
+  public void testConvertToSchedule_with_zero_withdraw() {
+    ScheduleRequest scheduleRequest = ScheduleRequest.builder()
+        .description("test")
+        .withdraw(BigDecimal.ZERO)
+        .recurringPattern(RecurringPattern.YEARLY)
+        .build();
+    ScheduleRequest.convertToSchedule(scheduleRequest);
+  }
+
+  @Test(expected = EzIllegalRequestException.class)
+  public void testConvertToSchedule_with_negative_withdraw() {
+    ScheduleRequest scheduleRequest = ScheduleRequest.builder()
+        .description("test")
+        .withdraw(BigDecimal.valueOf(-1L))
+        .recurringPattern(RecurringPattern.YEARLY)
+        .build();
+    ScheduleRequest.convertToSchedule(scheduleRequest);
+  }
+
+  @Test(expected = EzIllegalRequestException.class)
+  public void testConvertToSchedule_with_no_recurringPattern() {
+    ScheduleRequest scheduleRequest = ScheduleRequest.builder()
+        .description("test")
+        .withdraw(BigDecimal.ONE)
+        .build();
+    ScheduleRequest.convertToSchedule(scheduleRequest);
+  }
+
+  private void testConvertToScheduleWith(User user, Category category, Long updatingScheduleId) {
+    ScheduleRequest scheduleRequest = ScheduleRequest.builder()
+        .categoryId(1L)
+        .description("description")
+        .deposit(new BigDecimal("1000.00"))
+        .withdraw(null)
+        .recurringPattern(RecurringPattern.YEARLY)
+        .startDate(LocalDate.now())
+        .build();
+    Schedule schedule = null;
+    if (user == null && category == null && updatingScheduleId == null) {
+      schedule = ScheduleRequest.convertToSchedule(scheduleRequest);
+    } else if (user != null && category == null && updatingScheduleId == null) {
+      schedule = ScheduleRequest.convertToSchedule(scheduleRequest, user);
+    } else if (user != null && category != null && updatingScheduleId == null) {
+      schedule = ScheduleRequest.convertToSchedule(scheduleRequest, user, category);
+    } else if (user != null && category != null && updatingScheduleId != null) {
+      schedule = ScheduleRequest.convertToSchedule(scheduleRequest, user, category, updatingScheduleId);
+    }
+    if (updatingScheduleId == null) {
+      assertThat(schedule.getId()).isNull();
+    } else {
+      assertThat(schedule.getId()).isEqualTo(updatingScheduleId);
+    }
+    assertThat(schedule.getDescription()).isEqualTo(scheduleRequest.getDescription());
+    assertThat(schedule.getDeposit()).isEqualTo(scheduleRequest.getDeposit());
+    assertThat(schedule.getWithdraw()).isEqualTo(scheduleRequest.getWithdraw());
+    assertThat(schedule.getRecurringPattern()).isEqualTo(scheduleRequest.getRecurringPattern());
+    assertThat(schedule.getStartDate()).isEqualTo(scheduleRequest.getStartDate());
+    assertThat(schedule.getNextRecurringDate()).isEqualTo(scheduleRequest.getStartDate());
+    if (user == null) {
+      assertThat(schedule.getUser()).isNull();
```

Comment:
Shouldn't this be an exception instead if a schedule must belong to one user?

Reply:
https://github.com/mwkim0919/EZ/pull/4#discussion_r232442438